src/test/java/ru/training/at/hw3/tests/ExerciseFirst.java

```diff
@@ -0,0 +1,128 @@
+
+        //5.Assert that there are 4 items on the header
+        //section are displayed and they have proper texts
+        Header header = PageFactory.initElements(webDriver, Header.class);
+        waitActions.waitUntilCondition(webDriver -> header.getHomeHeader().isDisplayed());
+        softAssert.assertEquals(header.getHomeHeader().getText(),
+                prop.getProperty("homeHeader"));
+        waitActions.waitUntilCondition(webDriver -> header.getContactFormHeader().isDisplayed());
+        softAssert.assertEquals(header.getContactFormHeader().getText(),
+                prop.getProperty("contactFormHeader"));
+        waitActions.waitUntilCondition(webDriver -> header.getServiceHeader().isDisplayed());
+        softAssert.assertEquals(header.getServiceHeader().getText(),
+                prop.getProperty("serviceHeader"));
+        waitActions.waitUntilCondition(webDriver -> header.getMetalsAndColorsHeader().isDisplayed());
+        softAssert.assertEquals(header.getMetalsAndColorsHeader().getText(),
+                prop.getProperty("metalsAndColorsHeader"));
+
+        //6.Assert that there are 4 images on the Index Page and they are displayed
+        waitActions.waitUntilCondition(webDriver -> homePage.getImageFirstHomePage().isDisplayed());
+        softAssert.assertTrue(homePage.getImageFirstHomePage().isDisplayed());
+        waitActions.waitUntilCondition(webDriver -> homePage.getImageSecondHomePage().isDisplayed());
+        softAssert.assertTrue(homePage.getImageSecondHomePage().isDisplayed());
+        waitActions.waitUntilCondition(webDriver -> homePage.getImageThirdHomePage().isDisplayed());
+        softAssert.assertTrue(homePage.getImageThirdHomePage().isDisplayed());
+        waitActions.waitUntilCondition(webDriver -> homePage.getImageFourthHomePage().isDisplayed());
+        softAssert.assertTrue(homePage.getImageFourthHomePage().isDisplayed());
+
+        //7.Assert that there are 4 texts on the Index Page under icons and they have proper text
+        waitActions.waitUntilCondition(webDriver -> homePage.getTextFirstHomePage().isDisplayed());
+        softAssert.assertEquals(homePage.getTextFirstHomePage().getText(),
+                prop.getProperty("textFirstHomePage"));
+        waitActions.waitUntilCondition(webDriver -> homePage.getTextSecondHomePage().isDisplayed());
+        softAssert.assertEquals(homePage.getTextSecondHomePage().getText(),
+                prop.getProperty("textSecondHomePage"));
+        waitActions.waitUntilCondition(webDriver -> homePage.getTextThirdHomePage().isDisplayed());
+        softAssert.assertEquals(homePage.getTextThirdHomePage().getText(),
+                prop.getProperty("textThirdHomePage"));
+        waitActions.waitUntilCondition(webDriver -> homePage.getTextFourthHomePage().isDisplayed());
+        softAssert.assertEquals(homePage.getTextFourthHomePage().getText(),
+                prop.getProperty("textFourthHomePage"));
```

Comment:
Why do you do not combine text as list of web elements, extract its and compare two collection between each other?